src/dotnet/MLSearch.Service/Indexing/ChatIndexInitializer.cs

```diff
@@ -0,0 +1,50 @@
+using ActualChat.MLSearch.ApiAdapters.ShardWorker;
+
+namespace ActualChat.MLSearch.Indexing;
+
+internal interface IChatIndexInitializer
+{
+    ValueTask PostAsync(MLSearch_TriggerChatIndexingCompletion job, CancellationToken cancellationToken);
+}
+
+internal class ChatIndexInitializer(
+    IServiceProvider services,
+    ShardScheme shardScheme,
+    IShardIndexResolver<string> shardIndexResolver,
+    IChatIndexInitializerShard chatIndexInitializerShard,
+    ILogger<ChatIndexInitializer> log
+) : ShardWorker(services, shardScheme, nameof(ChatIndexInitializer)), IChatIndexInitializer
+{
+    private record DummyEvent() : IHasShardKey<string>
```

Comment:
- Not a blocker - 
Note: this must be solved with commands sent to a correct sharding scheme

Reply:
I agree with this, so I added todo item for the corresponding refactoring.